### ISSUE TITLE
'NoneType' object has no attribute 'build_dataset'

### ISSUE THREAD

**Issue by sxc95827** [Burhan-Q/ClassifyDefectMap_MixedWM38#3]
Hi Burhan,

There is an issue when I run your code. It seems that yolov8 has modified some interfaces.

Traceback (most recent call last):
  File "c:\Users\16119\Desktop\ClassifyDefectMap_MixedWM38-master\yolo_wafer_classification.py", line 13, in <module>
    model.trainer.build_dataset(img_path=data_dir) # loads data, but misses 'val_dir'
AttributeError: 'NoneType' object has no attribute 'build_dataset'

**Comment by Burhan-Q** [comment]
@sxc95827 yes you're right. This was a one-off project I did for an interview but I should refactor it to use a more up-to-date version of `ultralytics` 😆

**Comment by NG5156** [comment]
Hi Burhan,
When I calculate the confusion matrix,1 have a question,why your code is  accuracy = (tp[c] + 0) / (tp[c] + 0 + fp[c] + fn[c]) # zeros are True-Negatives, but there are none for this dataset ?
The data set consists of normal samples,  Why not use Accuracy =（TP+TN)/(TP+FP+TN+FN),

**Comment by Burhan-Q** [comment]
@NG5156 as I mentioned previously, this was a one-off project and needs updating/refactoring (including the calculation you mentioned). I believe this value _should_ be extractable from the confusion matrix returned.
